core/vm/evm.go

```diff
@@ -45,12 +45,20 @@ type (
 	// GetVRFFunc returns the nth block vrf in the blockchain
 	// and is used by the precompile VRF contract.
 	GetVRFFunc func(uint64) common.Hash
+	// Below functions are used by staking precompile / state transition
+	CreateValidatorFunc func(db StateDB, stakeMsg *stakingTypes.CreateValidator) error
```

Comment:
why removing create and edit validator ? that would be useful for the node runner to manage their validator via metamask

Reply:
> We may also remove CreateValidator and EditValidator

These cannot be removed because they are used in `state_transition.go` as `err = st.evm.CreateValidator(st.evm.StateDB, stkMsg)` and `err = st.evm.EditValidator(st.evm.StateDB, stkMsg)`; and the types need to be defined for use in the `evm` object.